<launch>
    <arg name="pcd_file" default="/home/orangepi/Documents/rospcd/dianguang-floor2-1-good_downsampled.pcd"/>
    <arg name="rviz" default="true" />

    <param name="takeoff_height" value="0.6" />

    <!-- global localization -->
    <include file="$(find uav_localization)/launch/localization_mid360.launch" />


    <!-- local Odometry -->
    <include file="$(find fast_lio)/launch/mapping_mid360.launch" >
        <arg name="rviz" value="false" />
    </include>


    <!-- global map -->
    <include file="$(find uav_planner)/launch/octomap_publisher.launch" >
        <arg name="pcd_file" value="$(arg pcd_file)"/>
    </include>


    <!-- global planner main -->
    <arg name="octomap_topic" default="/octomap_full" />
    <arg name="start_pose_topic" default="/start_pose" />
    <arg name="goal_pose_topic" default="/goal_pose" />
    <arg name="enable_obstacle_show" default="true" />
    <arg name="inflater_size" default="2" />

    <include file="$(find uav_planner)/launch/astar_planner.launch" >
        <arg name="octomap_topic" value="$(arg octomap_topic)" />
        <arg name="start_pose_topic" value="$(arg start_pose_topic)" />
        <arg name="goal_pose_topic" value="$(arg goal_pose_topic)" />
        <arg name="enable_obstacle_show" value="$(arg enable_obstacle_show)" />
        <arg name="inflater_size" value="$(arg inflater_size)" />
    </include>


    <!-- trace global odom to global path -->
    <node pkg="uav_planner" type="path_follower" name="path_follower" output="screen">
        <param name="goal_distance_threshold" value="0.3"/>
        <param name="path_step" value="2"/>
        <param name="publish_rate" value="5.0"/>
        <param name="odom_topic" value="/tf"/>
    </node>


    <!-- local planner main -->
    <include file="$(find ego_planner)/launch/single_run_in_exp.launch" />


    <!-- mission manager main -->
    <include file="$(find uav_px4_ctrl)/launch/mission_manger.launch" />
    

    <group if="$(arg rviz)">
	    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz_planner" 
              args="-d $(find uav_locatiation)/rviz/global_planner.rviz" />
	</group>
</launch>
